src/Optical/Emitters/Directions.jl

```diff
@@ -31,6 +31,10 @@ Constant(::Type{T} = Float64) where {T<:Real}
 struct Constant{T} <: AbstractDirectionDistribution{T}
     direction::Vec3{T}
 
+    function Constant(direction::T...) where{T<:Real}
```

Comment:
@alfredclwong does this cause allocations? It shouldn't but in the past using ... in functions has consistently caused allocations.

Reply:
We should change it anyway as `direction` has a well defined length. I think I've seen you make this change in another branch, though, so can we hold off doing it in this one?